web_analytics.go

```diff
@@ -0,0 +1,393 @@
+		return nil, ErrMissingWebAnalyticsRulesetID
+	}
+	uri := fmt.Sprintf("/accounts/%s/rum/v2/%s/rules", rc.Identifier, params.RulesetID)
+	res, err := api.makeRequestContext(ctx, http.MethodGet, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+	var r WebAnalyticsRulesResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result, nil
+}
+
+type DeleteWebAnalyticsRuleParams struct {
+	RulesetID string
+	RuleID    string
+}
+
+// DeleteWebAnalyticsRule deletes a Web Analytics Rule for a WebAnalyticsRuleset.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-delete-rule
+func (api *API) DeleteWebAnalyticsRule(ctx context.Context, rc *ResourceContainer, params DeleteWebAnalyticsRuleParams) (*string, error) {
+	if params.RulesetID == "" {
+		return nil, ErrMissingWebAnalyticsRulesetID
+	}
+	if params.RuleID == "" {
+		return nil, ErrMissingWebAnalyticsRuleID
+	}
+	uri := fmt.Sprintf("/accounts/%s/rum/v2/%s/rule/%s", rc.Identifier, params.RulesetID, params.RuleID)
+	res, err := api.makeRequestContext(ctx, http.MethodDelete, uri, nil)
+	if err != nil {
+		return nil, err
+	}
+	var r WebAnalyticsIDResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result.ID, nil
+}
+
+type UpdateWebAnalyticsRuleParams struct {
+	RulesetID string
+	RuleID    string
+	Rule      CreateWebAnalyticsRule
+}
+
+// UpdateWebAnalyticsRule lets you replace a Web Analytics Rule for a WebAnalyticsRuleset.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-update-rule
+func (api *API) UpdateWebAnalyticsRule(ctx context.Context, rc *ResourceContainer, params UpdateWebAnalyticsRuleParams) (*WebAnalyticsRule, error) {
+	if params.RulesetID == "" {
+		return nil, ErrMissingWebAnalyticsRulesetID
+	}
+	if params.RuleID == "" {
+		return nil, ErrMissingWebAnalyticsRuleID
+	}
+	uri := fmt.Sprintf("/accounts/%s/rum/v2/%s/rule/%s", rc.Identifier, params.RulesetID, params.RuleID)
+	res, err := api.makeRequestContext(ctx, http.MethodPut, uri, params.Rule)
+	if err != nil {
+		return nil, err
+	}
+	var r WebAnalyticsRuleResponse
+	err = json.Unmarshal(res, &r)
+	if err != nil {
+		return nil, fmt.Errorf("%s: %w", errUnmarshalError, err)
+	}
+	return &r.Result, nil
+}
+
+type ModifyWebAnalyticsRulesParams struct {
+	RulesetID string
+	// Rules is a list of rules to create if no id is specified, or modify if an existing id is specified.
+	Rules []CreateWebAnalyticsRule
+	// DeleteRules is a list of rule ids to be deleted
+	DeleteRules []string
+}
+
+// ModifyWebAnalyticsRules lets you create, updated, or delete multiple Web Analytics Rule for a WebAnalyticsRuleset.
+//
+// API reference: https://api.cloudflare.com/#web-analytics-modify-rules
+func (api *API) ModifyWebAnalyticsRules(ctx context.Context, rc *ResourceContainer, params ModifyWebAnalyticsRulesParams) (*WebAnalyticsRulesetRules, error) {
```

Comment:
this feels dicey; do we need it? can we stick with explicit CRUD operations?

Reply:
Yea, I think this was mainly for Dash functionality anyway.
We're probably good to remove it as it won't be used in Terraform anyway.